### PR TITLE
Future for calling overridden superclass methods

### PR DESCRIPTION
**Pull request by e-kayrakli**
Adds a new future for errors encountered calling an overridden method of a superclass from within the subclass.

### CODE REVIEW

**Comment by e-kayrakli** [comment]
@lydia-duncan -- any thoughts?

File: test/classes/engin/callSuperMethods.future [hunk 1]
```diff
@@ -0,0 +1,24 @@
+bug: superclass' methods cannot be invoked from subclass
+
+NOTES
+
+- Keyword "super" seems to parse and work in some cases but, it fails without any
+error message when used from inside the subclass
+
+- One woreking example for _super_ is when you use it outside the class
```
Comment (lydia-duncan) [inline test/classes/engin/callSuperMethods.future]: spellcheck: "woreking" -> "working".

File: test/classes/engin/callSuperMethods.chpl [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+  /*
+      bar runs if there is no foo that calls super.foo
+
+  proc bar() {
+    super.foo();
+    writeln("B.foo() called");
+  }
+  */
+  proc foo() {
```
Comment (lydia-duncan) [inline test/classes/engin/callSuperMethods.chpl]: My strategy is generally to make a few test cases for error messages - show the ones that work either in individual tests, or all in one test file that is separate from the error; and put each of the ways to trigger the error into their own file, so we can easily see if all have been fixed in one go or not.

**Comment by lydia-duncan** [comment]
My thoughts are that we need someone to write unit tests so we have a greater test coverage.  Also, tears.  So many tears.  None of that is your problem, though; aside from my two comments this looks fine to me

**Comment by lydia-duncan** [comment]
:+1:

**Pull request merged**